pyat/at/plot/resonances.py

```diff
@@ -0,0 +1,505 @@
+        "linewidth": numpy.mod(2, widthmod),
+        "label": "4s",
+    }
+    prop5s = {
+        "color": mypalettecolor[4],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "5s",
+    }
+    prop6s = {
+        "color": mypalettecolor[5],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "6s",
+    }
+    prop7s = {
+        "color": mypalettecolor[6],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "7s",
+    }
+    prop8s = {
+        "color": mypalettecolor[7],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "8s",
+    }
+    prop9s = {
+        "color": mypalettecolor[8],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "9s",
+    }
+    prop10s = {
+        "color": mypalettecolor[9],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "10s",
+    }
+    prop11s = {
+        "color": mypalettecolor[10],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "11s",
+    }
+    prop12s = {
+        "color": mypalettecolor[11],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "12s",
+    }
+    prop13s = {
+        "color": mypalettecolor[12],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "13s",
+    }
+    prop14s = {
+        "color": mypalettecolor[13],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "14s",
+    }
+    prop15s = {
+        "color": mypalettecolor[14],
+        "linestyle": mypalettestyle[1],
+        "linewidth": numpy.mod(1, widthmod),
+        "label": "15s",
+    }
+    # assemble default dictionary by normal or skew
+    propn = {
+        0: prop1n.copy(),
+        1: prop2n.copy(),
+        2: prop3n.copy(),
+        3: prop4n.copy(),
+        4: prop5n.copy(),
+        5: prop6n.copy(),
+        6: prop7n.copy(),
+        7: prop8n.copy(),
+        8: prop9n.copy(),
+        9: prop10n.copy(),
+        10: prop11n.copy(),
+        11: prop12n.copy(),
+        12: prop13n.copy(),
+        13: prop14n.copy(),
+        14: prop15n.copy(),
+    }
+    props = {
+        0: prop1s.copy(),
+        1: prop2s.copy(),
+        2: prop3s.copy(),
+        3: prop4s.copy(),
+        4: prop5s.copy(),
+        5: prop6s.copy(),
+        6: prop7s.copy(),
+        7: prop8s.copy(),
+        8: prop9s.copy(),
+        9: prop10s.copy(),
+        10: prop11s.copy(),
+        11: prop12s.copy(),
+        12: prop13s.copy(),
+        13: prop14s.copy(),
+        14: prop15s.copy(),
+    }
+    # assemble final dict
+    defaultlprop = {0: propn.copy(), 1: props.copy()}
```

Comment:
The default lifestyle directory could be initialized at the module level, so that it's done once for ever at import time. And are all the `copy()` necessary ?
However, it's a detail, I don't care too much…

Reply:
All call to copies where removed. They were not really needed.
Also, the dictionary was moved out of the function.